app/models/supermarket.rb

```diff
@@ -1,3 +1,13 @@
 class Supermarket < ApplicationRecord
   has_many :items
+
+  def unique_customer_names
```

Comment:
We are doing a little too much Ruby in here. We will want to use AR to query our database. 

If you were to add another association in here like
```ruby
has_many :customer, through: :items 
```
Then we can start right at `customer` and call `.distinct`
```ruby 
def unique_customer_names
 customers.distinct
end
```